src/node.cc

```diff
@@ -1380,7 +1381,19 @@ Isolate* NewIsolate(ArrayBufferAllocator* allocator, uv_loop_t* event_loop) {
   isolate->SetFatalErrorHandler(OnFatalError);
   isolate->SetAllowWasmCodeGenerationCallback(AllowWasmCodeGenerationCallback);
   v8::CpuProfiler::UseDetailedSourcePositionsForProfiling(isolate);
+  isolate->SetPromiseRejectCallback(task_queue::PromiseRejectCallback);
 
+  // NOTE: the following two callbacks currently CHECKs that the isolates
+  // have associated environments.
+  isolate->SetHostInitializeImportMetaObjectCallback(
```

Comment:
I am not entirely sure if we should do this here, or in `Environment` constructor, since these two callbacks do not handle the cases when the isolates' current context does not have Environments associated. Does it make sense to simply crash if the user tries to access `import()` or `import.meta` from a context without an associated environment? Or should we do nothing/return an empty Maybe instead?

Reply:
It’s tricky… I think for now crashing is okay (and the best we can do), and anything more would require interaction with embedders (which would be the main source for this not working).

Reply:
i don't think these ones should have been moved. they are set in response to core opting into handling them. if the handlers don't get attached, we shouldn't have these handlers enabled.

Reply:
I think I agree with @devsnek. Embedders can install their own after the call to `NewIsolate()`.